payjoin/src/core/send/v2/mod.rs

```diff
@@ -131,11 +114,26 @@ impl<'a> SenderBuilder<'a> {
         self,
         min_fee_rate: FeeRate,
     ) -> MaybeBadInitInputsTransition<SessionEvent, Sender<WithReplyKey>, BuildSenderError> {
-        let v1 = match self.0.build_non_incentivizing(min_fee_rate) {
+        self.v2_sender_from_v1(self.0.clone().build_non_incentivizing(min_fee_rate))
+    }
+
+    /// Helper function that takes a V1 sender build result and wraps it in a V2 Sender,
+    /// returning the appropriate state transition.
+    fn v2_sender_from_v1(
+        &self,
+        v1_result: Result<v1::Sender, BuildSenderError>,
+    ) -> MaybeBadInitInputsTransition<SessionEvent, Sender<WithReplyKey>, BuildSenderError> {
+        let mut v1 = match v1_result {
             Ok(inner) => inner,
             Err(e) => return MaybeBadInitInputsTransition::bad_init_inputs(e),
         };
 
+        // V2 senders may always ignore the receiver's `pjos` output substitution preference,
+        // because all communications with the receiver are end-to-end authenticated.
+        if self.0.output_substitution == OutputSubstitution::Enabled {
+            v1.output_substitution = OutputSubstitution::Enabled;
+        }
```

Comment:
I have w similar concern as @nothingmuch regarding this override. yes it *works* to override v1 fields with v2 preferences but it might be more legible if the hierarchy of v1/v2 preferences were evaluated at the specific version's code and  passed into a general function on sender::[v1::?]PsbtContext logic that took into account the net preference alone.

I had toyed around with tearing out PsbtContext from v1::Sender to fix #809 last week but couldn't justify the additional code. This change might change that balance in favor of the abstraction.

Reply:
The other problem with this is that it will undo the `always_disable_output_substitution` sender_builder method.